pkg/cmd/controller/route.go

```diff
@@ -13,20 +14,16 @@ import (
 	"k8s.io/klog/v2"
 	"k8s.io/kubernetes/pkg/api/legacyscheme"
 
+	openshiftcontrolplanev1 "github.com/openshift/api/openshiftcontrolplane/v1"
+
 	routecontrollers "github.com/openshift/openshift-controller-manager/pkg/cmd/controller/route"
 )
 
-func RunRouteControllerManager(ctx *ControllerContext, clientConfig *rest.Config) (bool, error) {
-	kubeClient, err := ctx.ClientBuilder.Client(infraIngressToRouteControllerServiceAccountName)
-	if err != nil {
-		return true, err
-	}
-	config := ctx.OpenshiftControllerConfig
-
+func RunRouteControllerManager(config *openshiftcontrolplanev1.OpenShiftControllerManagerConfig, kubeClient kubernetes.Interface, clientConfig *rest.Config) (bool, error) {
```

Comment:
this file is in a shared package. Should I move this to a route package so everything is also split on a package level?

Reply:
Yes, split this package.  I suggest creating `pkg/cmd/routecontroller`